wasmCloud/wasmCloud
# [FEAT] Downgrade claims to metadata, formalize policy service

## Feature or Problem
This PR:
- [ ] Downgrades embedded claims in actors from required to optional metadata
- [ ] Downgrades embedded claims in providers from required to optional metadata
- [ ] Reshapes the policy service request format to a more maintainable format with explicit policy request kinds and payloads. 
- [ ] Changed a few of the `par` internals to cache providers based on Host ID and provider ID, rather than link name and contract ID.

EDIT: I came back and added an OPA example 😄 

## Related Issues
Fixes #1441
Fixes #1104

## Release Information


## Consumer Impact
Implementations of policy servers will need to adjust their deserialization logic to receive the new requests.

## Testing


### Unit Test(s)


### Acceptance or Integration


### Manual Verification
Using a CLI policy service (`nats reply 'wasmcloud.policy' '{"permitted": true, "requestId": "abc123"}'`) I was able to see policy requests succeed and get cached. In the case where the operation was not permitted, error logs in the host report:
```
2024-03-01T18:34:56.058886Z ERROR wasmcloud_host::wasmbus: failed to scale actor actor_ref=file:///Users/brooks/github.com/wasmcloud/wasmcloud/tests/actors/rust/wrpc-ponger-component/build/wrpc_ponger_component_s.wasm actor_id=header err=Policy denied request to scale actor `abc123`: `None`
2024-03-01T18:36:18.559730Z ERROR wasmcloud_host::wasmbus: failed to handle invocation err=policy denied request to invoke component `foobar`: `None`
```
